junit5-api/src/main/java/org/junit/gen5/api/extension/ParameterResolver.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2015-2016 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.junit.gen5.api.extension;
+
+import static org.junit.gen5.commons.meta.API.Usage.Experimental;
+
+import java.lang.reflect.Parameter;
+import java.util.Optional;
+
+import org.junit.gen5.commons.meta.API;
+
+/**
+ * {@code ParameterResolver} defines the API for {@link Extension Extensions}
+ * that wish to dynamically resolve {@linkplain Parameter parameters} at runtime.
+ *
+ * <p>If a {@link org.junit.gen5.api.Test @Test},
+ * {@link org.junit.gen5.api.BeforeEach @BeforeEach},
+ * {@link org.junit.gen5.api.AfterEach @AfterEach},
+ * {@link org.junit.gen5.api.BeforeAll @BeforeAll}, or
+ * {@link org.junit.gen5.api.AfterAll @AfterAll} method accepts a parameter,
+ * the parameter must be resolved at runtime by a {@code ParameterResolver}.
+ *
+ * <p>Implementations must provide a no-args constructor.
+ *
+ * @since 5.0
+ */
+@API(Experimental)
+public interface ParameterResolver extends Extension {
+
+	/**
+	 * Determine if this resolver supports resolution of the given {@link Parameter}
+	 * for the supplied {@code target} and {@link ExtensionContext}.
+	 *
+	 * <p>The {@link java.lang.reflect.Method} or {@link java.lang.reflect.Constructor}
+	 * in which the {@code parameter} is declared can be retrieved via
+	 * {@link Parameter#getDeclaringExecutable()}.
```

Comment:
Not until you told me a few days ago. 😉

I just meant it's good that you documented it here. 🙂